GeometricImgTransform remap and remapImg are broken on MacOS

`stack build` on MacOS blows up with:

```
    [75 of 77] Compiling OpenCV.ImgProc.GeometricImgTransform ( .stack-work/dist/x86_64-osx/Cabal-2.2.0.1/build/OpenCV/ImgProc/GeometricImgTransform.hs, .stack-work/dist/x86_64-osx/Cabal-2.2.0.1/build/OpenCV/ImgProc/GeometricImgTransform.o )

    /Users/dtapley/dev/haskell-opencv/opencv//var/folders/4v/b2l26sy50g5_wyllvmgt_4l58r0b68/T/ghc49301_0/ghc_948.cpp:188:15: error:
         error: expected expression
                    , {}
                      ^
        |
    188 |             , {}
        |               ^
    1 error generated.
    `gcc' failed in phase `C Compiler'. (Exit code: 1)
```

--- a/opencv/src/OpenCV/ImgProc/GeometricImgTransform.hsc
+++ b/opencv/src/OpenCV/ImgProc/GeometricImgTransform.hsc
@@ -547,11 +547,12 @@ remap src mapping interpolationMethod borderMode = unsafeWrapException $ do
       withPtr mapping $ \mappingPtr ->
       withPtr borderValue $ \borderValuePtr ->
         [cvExcept|
+          cv::Mat map2;
           cv::remap
             ( *$(Mat * srcPtr)
             , *$(Mat * dstPtr)
             , *$(Mat * mappingPtr)
-            , {}
+            , map2
             , $(int32_t c'interpolation)
             , $(int32_t c'borderMode)
             , *$(Scalar * borderValuePtr)

